packages/Parallel/WorkerRunner.php

```diff
@@ -97,7 +97,7 @@ public function run(Encoder $encoder, Decoder $decoder, Configuration $configura
 
                     if ($errorAndFileDiffs[Bridge::SYSTEM_ERRORS] !== []) {
                         $this->invalidateFile($file);
-                    } elseif (! $configuration->isDryRun()) {
+                    } else {
```

Comment:
this require e2e test for:

- after multiple --dry-run keep show the diff, run apply changes to still apply the change.

For note: I think if you want current e2e timeout be deleted, that need to be replaced with "real crash use case", eg: can create a custom rule in e2e that must be crash, eg: echo dump($If_->else) node which the else still null so that will always crash.

Reply:
@samsonasik I have opened a new PR [Cache unchanged files on dry run v2](https://github.com/rectorphp/rector-src/pull/4281 ), it seems that all the tests are passing now. It looks like that [this rework](https://github.com/rectorphp/rector-src/commit/c5b17183be766c3abf9535f602c27d8f43a3e151#diff-63e3c3ba9e64088f30e6fad997198e74bebdb51dc6b551078436f3395aad1610) of the timeout test fixed the issue. Additionaly, I have updated the `e2e_consecutive_changes.yaml` to include two dry runs, covering the test case you mentioned. Please let me know if this is sufficient for the PR to get merged, thanks!